any_member.hpp

```diff
@@ -639,10 +633,23 @@ void MemberSymbolTable<ClassType>::ascribe
             >::value
         );
 
-    ClassType* class_object = static_cast<ClassType*>(this);
-    map_.insert(member_pair_type(s, any_member<ClassType>(class_object, p2m)));
-    // TODO ?? This would appear to be O(N^2).
     auto i = std::lower_bound(member_names_.begin(), member_names_.end(), s);
+
+    ClassType* class_object = static_cast<ClassType*>(this);
+
+    // Using the usual insert() method relies on vector elements being
+    // assignable in C++98, but this is not the case for any_member<>, so we
```

Comment:
This should work just fine in C++17 however, if `any_member` is movable which, AFAICS, it should be. Could you please make it so and get rid of using `swap()` here?